packages/ra-core/src/routing/useGetPathForRecord.stories.tsx

```diff
@@ -0,0 +1,182 @@
+};
+
+const EditLink = () => {
+    const path = useGetPathForRecord({ link: 'edit' });
+    return path ? <Link to={path}>Edit</Link> : <span>Edit no link</span>;
+};
+
+const ShowLink = () => {
+    const path = useGetPathForRecord({ link: 'show' });
+    return path ? <Link to={path}>Show</Link> : <span>Show no link</span>;
+};
+
+const InferredLink = () => {
+    const path = useGetPathForRecord();
+    return path ? <Link to={path}>Link</Link> : <span>No link</span>;
+};
+
+export const NoAuthProvider = () => (
+    <TestMemoryRouter>
+        <CoreAdminContext>
+            <ResourceContextProvider value="posts">
+                <RecordContextProvider value={{ id: 123 }}>
+                    <div style={{ display: 'flex', gap: 2 }}>
+                        <EditLink />
+                        <ShowLink />
+                    </div>
+                </RecordContextProvider>
+            </ResourceContextProvider>
+        </CoreAdminContext>
+    </TestMemoryRouter>
+);
+
+export const InferredEditLink = () => (
+    <TestMemoryRouter>
+        <CoreAdminContext>
+            <ResourceContextProvider value="posts">
+                <ResourceDefinitionContextProvider
+                    definitions={{
+                        posts: { name: 'posts', hasEdit: true, hasShow: false },
+                    }}
+                >
+                    <RecordContextProvider value={{ id: 123 }}>
+                        <div style={{ display: 'flex', gap: 2 }}>
+                            <InferredLink />
+                        </div>
+                    </RecordContextProvider>
+                </ResourceDefinitionContextProvider>
+            </ResourceContextProvider>
+        </CoreAdminContext>
+    </TestMemoryRouter>
+);
+
+export const InferredShowLink = () => (
+    <TestMemoryRouter>
+        <CoreAdminContext>
+            <ResourceContextProvider value="posts">
+                <ResourceDefinitionContextProvider
+                    definitions={{
+                        posts: { name: 'posts', hasEdit: false, hasShow: true },
+                    }}
+                >
+                    <RecordContextProvider value={{ id: 123 }}>
+                        <div style={{ display: 'flex', gap: 2 }}>
+                            <InferredLink />
+                        </div>
+                    </RecordContextProvider>
+                </ResourceDefinitionContextProvider>
+            </ResourceContextProvider>
+        </CoreAdminContext>
+    </TestMemoryRouter>
+);
+
+export const AccessControlWithLinkTypeProvided = ({
+    authProvider = {
+        login: () => Promise.resolve(),
+        logout: () => Promise.resolve(),
+        checkAuth: () => Promise.resolve(),
+        checkError: () => Promise.resolve(),
+        getPermissions: () => Promise.resolve(),
+        canAccess: ({ action }) =>
+            new Promise(resolve => setTimeout(resolve, 300, action === 'edit')),
```

Comment:
tests will be slow because of that. Can you use a suspended Promise in tests instead?

Reply:
I based the test on another story